### ISSUE TITLE
[BUG] App drawer has bigger margins when grid settings of homescreen are changed

### ISSUE THREAD

**Issue by dserranoc** [Deletescape-Media/Lawnchair#4260]
### Describe the bug

When the grid settings of the homescreen are at default values, there are no issues with either the homescreen or the app drawer. If I change the homescreen grid to 5 columns and 7 rows, the configuration is applied successfully. However, upon opening the app drawer, the app icons are "compacted," leaving more margin on both sides.

Edit: Removing the search bar of the homescreen dock "resolves" the issue.

### Steps to reproduce

1. Open the homescreen settings
2. Click on 'Homescreen grid'
3. Set it to 5 columns and 7 rows
4. Go to the homescreen.
5. The new config are applied, but open the app drawer.
6. See the error
7. (Optional) Disable the search bar of homescreen in dock settings.
8. Now the issue has been "resolved"

### Expected behavior

The app drawer does not have to be affected by the homescreen grid change. Even if it were a unified setting between the homescreen and the app drawer, the latter should not have to be broken.

### Screenshots

Settings:
![Screenshot_20240410_181234_Lawnchair](https://github.com/LawnchairLauncher/lawnchair/assets/21318401/dc85762c-a371-493a-abfa-18c5815fb5b4)

Homescreen:
![Screenshot_20240410_181241_Lawnchair](https://github.com/LawnchairLauncher/lawnchair/assets/21318401/b7d8c038-a1b1-4c39-92b3-8d11bf0b0fa7)

App drawer:
![Screenshot_20240410_181245_Lawnchair](https://github.com/LawnchairLauncher/lawnchair/assets/21318401/440d063c-0781-4545-a34b-53a0d70ca3fb)

![1712765902261](https://github.com/LawnchairLauncher/lawnchair/assets/21318401/7f1d6d5e-48b5-4e53-8358-99456ff3f1e7)

### Device information

- Device: Samsung Galaxy S23 and Poco X3 Pro
- OS: Android 14 with One UI 6.1 / MIUI 14.0.2
- App version: 14.0.0 Beta 2

### Additional context

Tried in two devices with the same result. It is a clean install.

**Comment by naiyerasif** [comment]
Are you restoring the settings from a backup before you do this change? With clean setup, this stopped happening for me.

**Comment by dserranoc** [comment]
No, in both phones it is a clean install. I add this info in the additional context.

**Comment by dserranoc** [comment]
I've found a workaround but it requires disabling the homescreen search bar.
